### PR TITLE
resolve KeyError: 'PDSH_SSH_ARGS_APPEND'

### PR DESCRIPTION
**Pull request by Lzhang-hub**
when start job with `deepspeed --hostfile hostfile --master_addr $MASTER_IP --ssh_port 20023 src/train_bash.py `

get error: KeyError: 'PDSH_SSH_ARGS_APPEND' in https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/launcher/multinode_runner.py#L77

because PDSH_SSH_ARGS_APPEND not in environment.

### CODE REVIEW

**Comment by loadams** [comment]
Thanks @Lzhang-hub - have you tested that this change works for you?

File: deepspeed/launcher/multinode_runner.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def name(self):
 
     def get_cmd(self, environment, active_resources):
         environment['PDSH_RCMD_TYPE'] = 'ssh'
-        if self.args.ssh_port is not None:  # only specify ssh port if it is specified
-            environment["PDSH_SSH_ARGS_APPEND"] += f" -p {self.args.ssh_port}"
+        if self.args.ssh_port is not None:   # only specify ssh port if it is specified
```
Comment (loadams) [inline deepspeed/launcher/multinode_runner.py:76]: FYI @Lzhang-hub - can you remove this added space on line 76?  Its causing the formatting checks to fail.

**Comment by Lzhang-hub** [comment]
> Thanks @Lzhang-hub - have you tested that this change works for you?
yes, I had test

File: deepspeed/launcher/multinode_runner.py (hunk 1 above)
Comment (Lzhang-hub) [reply deepspeed/launcher/multinode_runner.py:76]: ok

**Comment by Lzhang-hub** [comment]
> Thanks @Lzhang-hub - have you tested that this change works for you?

yes, I had tested.

File: deepspeed/launcher/multinode_runner.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def name(self):
     def get_cmd(self, environment, active_resources):
         environment['PDSH_RCMD_TYPE'] = 'ssh'
         if self.args.ssh_port is not None:  # only specify ssh port if it is specified
-            environment["PDSH_SSH_ARGS_APPEND"] += f" -p {self.args.ssh_port}"
+            environment["PDSH_SSH_ARGS_APPEND"] = f"{environment.get('PDSH_SSH_ARGS_APPEND', '')} -p {self.args.ssh_port}"
```
Comment (loadams) [inline deepspeed/launcher/multinode_runner.py:77]: @Lzhang-hub - you'll need to run the pre-commit hooks/formatter on this code to get it to pass the CI checks

Comment (Lzhang-hub) [reply deepspeed/launcher/multinode_runner.py]: ok,it passed

**Review by loadams — APPROVED** [review]

**Pull request merged**